### PR TITLE
Add UI

### PR DESCRIPTION
**Pull request by Fikerte-T**
- Build interactive console app (school library)
- Run a program using the command line

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      puts 'Student created successfully'
+    when 2
+      @person.push(create_teacher)
+      puts 'Teacher created successfully'
+    end
+  end
+
+  def create_student
+    puts 'Classroom: '
+    classroom = gets.chomp
```
Comment (SajjadAhmad14) [inline app.rb]: Please remove the option to input `classroom` when a user selects option `3` and wants to create a new `Student`. The user should only enter the name, age, and parent permission. Kindly see the screenshot below.

classroom

File: main [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env ruby
+require './app'
+
```
Comment (SajjadAhmad14) [inline main]: Please add the `.rb` extension as this is also a Ruby file. The project does not run without adding an extension to this file.

Comment (SajjadAhmad14) [inline main]: Please fix the linter checks as all the checks are failing on PR.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Fikerte, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: main (hunk 2 above)
Comment (Fikerte-T) [reply main]: Hello @SajjadAhmad14,
I am on Windows and this bash command is not working.

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    puts 'Name: '
+    name = gets.chomp.to_s
+    puts 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'y'
+      Student.new(classroom, age, name)
+    when 'n'
+      Student.new(classroom, age, name, parent_permission: false)
+    else
```
Comment (elisha2kyakpo1) [inline app.rb]: - [ ] Let's fix the issue requested by the previous reviewer, a student shouldn't have a classroom as a its parameter.

File: student.rb [hunk 4]
```diff
@@ -5,7 +5,7 @@ class Student < Person
   attr_reader :classroom
 
   def initialize(classroom, age, name = 'Unknown', parent_permission: true)
-    super(age, name, parent_permission)
```
Comment (elisha2kyakpo1) [inline student.rb]: - [ ] Same here, a classroom shouldn't be one the parameters in the initialize method

File: app.rb [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    options
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def perform_switch(choice)
+    case choice
+    when 1
+      list_books
+    when 2
+      list_people
```
Comment (elisha2kyakpo1) [inline app.rb:38]: - [ ] Let's work on fixing this issue in the image below. When I create a student and receive a successful message, then I try to list the created `list_people` I find that a person is not properly created.

student

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi @Fikerte-T, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

**I advise you to take your time and fix all the highlighted issues before submitting for review**

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'y'
+      Student.new(age, name)
+    when 'n'
+      Student.new(age, name, parent_permission: false)
+    else
+      puts 'Please enter Y/N[y/n]'
+      create_student
+    end
```
Comment (elisha2kyakpo1) [inline app.rb]: - [ ] I faced the error in the image below when I wanted to create a student.

second-review

File: student.rb [hunk 7]
```diff
@@ -4,8 +4,8 @@
 class Student < Person
   attr_reader :classroom
 
-  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
-    super(age, name, parent_permission)
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission:)
     @classroom = classroom
```
Comment (elisha2kyakpo1) [inline student.rb]: - [ ] Let's remove the colon at the end of parent_permission, it is rising an error. I would advise you to test the project before submitting it for review

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi @Fikerte-T, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 6 above)
Comment (Fikerte-T) [reply app.rb]: I don't know what's happening on your side, but its working perfectly fine on my side

File: student.rb (hunk 7 above)
Comment (Fikerte-T) [reply student.rb]: I don't know what's happening on your side, but its working perfectly fine on my side

**Review by Dipeshtwis — APPROVED** [review]
# Hi @Fikerte-T, 

### Status: Approved ✔️

:+1:  You have implemented the suggested changes.

Your project is complete!  There are some optional changes I've suggested to you to make your project best, other than that nothing else to say.
Congratulations! 🎉 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- In the README file, to run the project, it should be `ruby main.rb` and not `ruby my_list.rb`. Please fix it.


**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
-----

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Dipesh Kumar_**

------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**